include/odp/api/spec/packet.h

```diff
@@ -388,6 +416,26 @@ odp_event_t odp_packet_to_event(odp_packet_t pkt);
 void odp_packet_to_event_multi(const odp_packet_t pkt[], odp_event_t ev[],
 			       int num);
 
+/**
+ * Recover individual fragments from a packet with incomplete reassembly
+ *
+ * In case of incomplete reassembly, the packet would consist of multiple
+ * individual fragments. Recover all such fragments from the packet.
+ *
+ * @param      pkt   Packet handle to recover frags.
+ *                   Post successful recovery of fragments, the packet would be
+ *                   freed.
+ * @param[out] frags Packet handle array for output. The size of this array must
+ *                   be larger than `odp_reassembly_config_t.ip.max_nb_frags`
+ *                   configuration.
+ * @param[out] res   Handle to store the result from the recovery operation
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_packet_recover_frags(odp_packet_t pkt, odp_packet_t frags[],
+			     odp_packet_reassem_recover_res_t *res);
```

Comment:
"Recover" seems to imply that we get the original fragments, but maybe we do not want to promise that. Maybe part of the fragments are already partially reassembled from original smaller fragments.

Perhaps the function should be called odp_reass_result() (c.f. crypto, ipsec and comp APIs)`?

Maybe something like this would be a better wording for the description:
```
/*
 * Get partial reassembly result from a packet
 *
 * In case of incomplete reassembly, a packet carries information on
 * the time already used for the reassembly attempt and one or more
 * fragments that do not necessarily correspond to the original
 * received fragments.
 *
 * This function may be called only if the reassembly status of a packet
 * is ODP_PACKET_REASS_INCOMPLETE.
 *
 * @param      pkt   Incompletely reassembled packet. The packet will
 *		     be consumed if the function succeeds.
 * @param[out] frags Packet handle array for output. The size of this array must
 *                   be at least `odp_reassembly_config_t.ip.max_nb_frags`.
 * @param[out] res   Pointer to result structure
 *
 * @retval 0 on success
 * @retval <0 on failure
 */
```

Reply:
Agreed. These datapath functions were drafted as packet APIs. But with the change in function name, may be these can be presented as reassembly APIs (move all code additions to to reassembly.h file). Do you agree?